server/repository/src/db_diesel/changelog/changelog.rs

```diff
@@ -158,13 +177,15 @@ impl<'a> ChangelogRepository<'a> {
     }
 }
 
-type BoxedChangelogQuery = IntoBoxed<'static, changelog_deduped::table, DBType>;
+type BoxedChangelogQuery =
+    IntoBoxed<'static, LeftJoin<changelog_deduped::table, name_link::table>, DBType>;
 
 fn create_filtered_query<'a>(
     earliest: u64,
     filter: Option<ChangelogFilter>,
 ) -> BoxedChangelogQuery {
     let mut query = changelog_deduped::dsl::changelog_deduped
+        .left_join(name_link::table)
```

Comment:
I've been a bit confused by how we have both `some_table_dsl` and `some_table::table` imported in files when they directly overlap, and I think I understand that it is really a confused use of `dsl`

[Reading docs for table::dsl](https://docs.rs/diesel/1.4.8/diesel/macro.table.html#dsl):

>This simply re-exports the table, renamed to the same name as the module, and each of the columns. This is useful to glob import when you’re dealing primarily with one table, to allow writing `users.filter(name.eq("Sean"))` instead of `users::table.filter(users::name.eq("Sean"))`.
 
So... it's really fine just to use the "table", dsl is just for convenience if you want to just do `use ...dsl::*` and lucky enough that doesn't conflict in the scope. `changelog_deduped::dsl::changelog_deduped` should really just be `changelog_deduped::table`.

Reply:
With this new clarity I have made a commit clearing out a whole lot of dsl cruft in this repo 😃 9a9223ae7f488bf4a9f7fea9e10369f8f916245d

Reply:
I never thought about it, but this makes sense :) Think we started from a simple example and then run into problems when using two conflicting DSLs at the same time. For which reason we ended up where we are.

Reply:
I guess you can't have a field named table though, or you'd conflict on `my_table::table`?

Reply:
Huh... is that why we have:

https://github.com/msupply-foundation/open-msupply/blob/65f0950a895668b51fa763b7bfffe1f3457ad12b/server/repository/src/db_diesel/name_row.rs#L14-L17

Reply:
Yes, and the rename of `type` is because it's a Rust key word....